### PR TITLE
fix some minor issues:

### PR DESCRIPTION
**Pull request by YaoZengzeng**
1.make the path comparison of Makefile more robust

Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### CODE REVIEW

**Comment by ymesika** [comment]
Please rebase and see if it's still relevant

**Comment by YaoZengzeng** [comment]
@ymesika This PR will only be leaved to solve the Makefile issue.

There seems a misuse with the way we get the registries and handle it.

I'll try to fix it in another PR.

File: Makefile [hunk 1]
```diff
@@ -233,7 +233,7 @@ ${ISTIO_BIN}/have_go_$(GO_VERSION_REQUIRED):
 # Ensure expected GOPATH setup
 .PHONY: check-tree
 check-tree:
-	@if [ "$(ISTIO_GO)" != "$(GO_TOP)/src/istio.io/istio" ]; then \
+	@if [ ! "$(ISTIO_GO)" -ef "$(GO_TOP)/src/istio.io/istio" ]; then \
 		echo Not building in expected path \'GOPATH/src/istio.io/istio\'. Make sure to clone Istio into that path. Istio root=$(ISTIO_GO), GO_TOP=$(GO_TOP) ; \
```
Comment (ymesika) [inline Makefile]: This is good opportunity to add `""` to echo variables:
```
... Make sure to clone Istio into that path. Istio root="$(ISTIO_GO)", GO_TOP="$(GO_TOP)" ; \
```
Can you please fix it too?

File: pilot/pkg/serviceregistry/aggregate/controller.go [hunk 2]
```diff
@@ -56,9 +56,7 @@ func (c *Controller) AddRegistry(registry Registry) {
 	c.storeLock.Lock()
 	defer c.storeLock.Unlock()
 
-	registries := c.registries
-	registries = append(registries, registry)
-	c.registries = registries
+	c.registries = append(c.registries, registry)
```
[the remark this answers is not in the document]
Comment (YaoZengzeng) [reply pilot/pkg/serviceregistry/aggregate/controller.go]: @ymesika Updated

File: Makefile (hunk 1 above)
Comment (YaoZengzeng) [reply Makefile]: @ymesika Updated.

**Pull request merged**